src/mjolnir/ferry_connections.cc

```diff
@@ -196,8 +206,9 @@ uint32_t ShortestPath(const uint32_t start_node_idx,
     uint16_t path_access = baldr::kVehicularAccess;
     while (true) {
       // Get the edge between this node and the predecessor
-      uint32_t idx = node_labels[label_idx].node_index;
-      uint32_t pred_node = node_labels[label_idx].pred_node_index;
+      const NodeLabel& node_lab = node_labels[last_label_idx];
```

Comment:
there was bug before where we were taking `label_idx` which was left over from the expansion. however, `label_idx` could be pointing to some random stuff in the adjacency list, just before the expansion was finished, and that label wouldn't even be part of the path.. so here we initialized the path forming with the wrong label